velox/connectors/hive/storage_adapters/s3fs/CMakeLists.txt

```diff
@@ -14,11 +14,17 @@
 
 # for generated headers
 
-add_library(velox_s3fs S3FileSystem.cpp S3Util.cpp)
-target_include_directories(velox_s3fs PUBLIC ${AWSSDK_INCLUDE_DIRS})
-target_link_libraries(velox_s3fs Folly::folly ${AWSSDK_LIBRARIES})
+add_library(velox_s3fs RegisterS3FileSystem.cpp)
+if(VELOX_ENABLE_S3)
+  target_sources(velox_s3fs PRIVATE S3FileSystem.cpp S3Util.cpp)
 
-if(${VELOX_BUILD_TESTING})
-  add_subdirectory(tests)
-  add_subdirectory(benchmark)
+  target_include_directories(velox_s3fs PUBLIC ${AWSSDK_INCLUDE_DIRS})
+  target_link_libraries(velox_s3fs Folly::folly ${AWSSDK_LIBRARIES})
```

Comment:
I looked into the aws sdk cmake and it seems that it's a bit special and has no single toplevel target to link against... so there seems to be no better way but otherwise please link against targets directly and don't use vars as they default to empty string which causes debugging nightmares ^^

It should also be fine to skip include dir as `${AWSSDK_LIBRARIES}` should contain targets that come with that info already (have not tested that thought)

Reply:
@assignUser I will address your comments in a separate PR as they are independent to this change.

Reply:
I'd be  happy if you found a way to use targets directly but I understand that you have to work with aws sdk gives you so I would be ok with keeping it as is in this case :)